tests/cloner_test.go

```diff
@@ -574,6 +575,75 @@ var _ = Describe("all clone tests", func() {
 						string(dv.Status.Progress) == "N/A"
 				}, timeout, pollingInterval).Should(BeTrue(), "DV Should succeed with storage.prePopulated==pvcName")
 			})
+
+			DescribeTable("Should clone with empty volume size without using size-detection pod",
+				func(sourceVolumeMode v1.PersistentVolumeMode, targetVolumeMode v1.PersistentVolumeMode) {
+					// When cloning without defining the target's storage size, the source's size can be attainable
+					// by different means depending on the clone type and the volume mode used.
+					// Either if "block" is used as volume mode or smart/csi cloning is used as clone strategy,
+					// the value is simply extracted from the original PVC's spec.
+					if cloneType == "network" && sourceVolumeMode == v1.PersistentVolumeFilesystem {
+						Skip("Clone strategy and volume mode combination requires of size-detection pod")
+					}
+
+					if sourceVolumeMode != targetVolumeMode && cloneType == "csivolumeclone" {
```

Comment:
Don't need this skip csi clone/snapshot clone only works if the source and target volume mode are the same, and if they both use the same storage class.